src/API/Baggage/Propagation/Parser.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\API\Baggage\Propagation;
+
+use function explode;
+use OpenTelemetry\API\Baggage\BaggageBuilderInterface;
+use OpenTelemetry\API\Baggage\Metadata;
+use function str_contains;
+use function str_replace;
+use function trim;
+
+final class Parser
+{
+    private const EXCLUDED_KEY_CHARS = [' ', '(', ')', '<', '>', '@', ',', ';', ':', '\\', '"', '/', '[', ']', '?', '=', '{', '}'];
+    private const EXCLUDED_VALUE_CHARS = [' ', '"', ',', ';', '\\'];
+
+    /** @readonly */
+    private string $baggageHeader;
+
+    public function __construct(string $baggageHeader)
+    {
+        $this->baggageHeader = $baggageHeader;
+    }
+
+    /**
+     * @todo: Make this algorithm more robust/performant.
+     */
+    public function parseInto(BaggageBuilderInterface $baggageBuilder): void
+    {
+        foreach (explode(',', $this->baggageHeader) as $baggageString) {
+            if (empty(trim($baggageString))) {
+                continue;
+            }
+
+            $explodedString = explode(';', $baggageString, 2);
+
+            $keyValue = trim($explodedString[0]);
+
+            if (empty($keyValue) || !str_contains($keyValue, '=')) {
```

Comment:
Is there going to be a time where the second conditional here rings true?  Should we potentially add additional tests for these?

Reply:
Also, phan was complaining about `str_contains()` without a polyfill:
https://github.com/open-telemetry/opentelemetry-php/pull/437/checks?check_run_id=3798402726#step:9:48

Reply:
It's handled via the `valid complex value with some invalid` case. When it splits, the third baggage value, `a sdf9asdf-alue1` ends up being `$keyValue` which isn't valid since there is no key.

The spec does say:

> Parsers MUST NOT assume that the equal sign is only used to separate key and value.

So might not hurt to add some additional cases where there are extraneous `=`.